packages/connectivity/src/scp-cf/authorization-header.ts

```diff
@@ -184,9 +161,10 @@ function getProxyRelatedAuthHeaders(
     destination.proxyType === 'OnPremise' &&
     destination.authentication === 'NoAuthentication'
   ) {
-    return legacyNoAuthOnPremiseProxy(destination) as any;
+    throw Error(
+      'NoAuthentication is no longer supported for OnPremise connections'
```

Comment:
Perhaps I would repharse this to a more active form: OnPremise connections are not possible with NoAuthentication. Please select a supported auth flow e.g. principal propagation or basic.

Reply:
True, I'll also add the removed test case back and instead test for the error message

Reply:
Or is this test case too trivial?